### ISSUE TITLE
Trying (i.e. Failing) to Use Lilypond

### ISSUE THREAD

**Issue by BetazoidMan** [jperon/lyluatex#293]
Hello,
I am trying to use Lilypond with LuaLaTeX so that I can use GregorioTeX at the same time. I have been unable to make progress trying to diagnose this myself. I will be appreciative of any aid you can give. 
System: Ubuntu 22.04.1 LTS (Kubuntu flavor)
Environment: TeXworks 0.6.7

Input:
```
\documentclass{article}

\usepackage{lyluatex}

\begin{document}
\begin{lilypond}[quote,fragment,staffsize=26]
c' d' e' f' g'2 g'2
\end{lilypond}

\medskip
\end{document}
``` 

Console Output
``` 
This is LuaHBTeX, Version 1.14.0 (TeX Live 2022/dev/Debian) 
 system commands enabled.
(./untitled-3.tex
LaTeX2e <2021-11-15> patch level 1
[\directlua]:1: module 'luaotfload-main' not found:
	no field package.preload['luaotfload-main']
	[kpse lua searcher] file not found: 'luaotfload-main'
	[kpse C searcher] file not found: 'luaotfload-main'
Error in luaotfload: reverting to OT1 L3 programming layer <2022-01-21>
(/usr/share/texlive/texmf-dist/tex/latex/base/article.cls
Document Class: article 2021/10/04 v1.4n Standard LaTeX document class
(/usr/share/texlive/texmf-dist/tex/latex/base/size10.clo
(/usr/share/texmf/tex/latex/lm/ot1lmr.fd)))

! LaTeX Error: File `lyluatex.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: 
(/usr/share/texlive/texmf-dist/tex/latex/l3backend/l3backend-luatex.def)
(./untitled-3.aux) (/usr/share/texlive/texmf-dist/tex/latex/base/ts1cmr.fd)

! LaTeX Error: Environment lilypond undefined.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.6 \begin{lilypond}
                  [quote,fragment,staffsize=26]
? 

! LaTeX Error: \begin{document} ended by \end{lilypond}.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.8 \end{lilypond}
                
? 
[1{/var/lib/texmf/fonts/map/pdftex/updmap/pdftex.map}] (./untitled-3.aux))
 368 words of node memory still in use:
   3 hlist, 1 vlist, 1 rule, 2 glue, 48 glue_spec, 1 write nodes
   avail lists: 2:12,3:4,4:1,5:24,6:2,7:138,9:18,11:4
{/usr/share/texmf/fonts/enc/dvips/lm/lm-rm.enc}</usr/share/texmf/fonts/type1/pu
blic/lm/lmr10.pfb>
Output written on untitled-3.pdf (1 page, 27174 bytes).
SyncTeX written on untitled-3.synctex.gz.
Transcript written on untitled-3.log.
``` 
Log information attached in [log file.](https://github.com/jperon/lyluatex/files/9739839/untitled-3.log)

**Comment by rpspringuel** [comment]
> `! LaTeX Error: File `lyluatex.sty' not found.`

This message is saying that lyluatex is not installed properly.  How did you install it?

**Comment by BetazoidMan** [comment]
I tried to follow the instructions using `tlmgr install lyluatex` but I didn't realize that this does not work with Debian. 
For the time being, I have decided to go with the "For a single document" option. Now I'm getting a different litany of errors. 
Here is my input now: 

```
\documentclass{article}
\usepackage{lyluatex}

\begin{document}
\end{document}
```

Console output: 
```
This is LuaHBTeX, Version 1.14.0 (TeX Live 2022/dev/Debian) 
 system commands enabled.
(./untitled-3.tex
LaTeX2e <2021-11-15> patch level 1
[\directlua]:1: module 'luaotfload-main' not found:
	no field package.preload['luaotfload-main']
	[kpse lua searcher] file not found: 'luaotfload-main'
	[kpse C searcher] file not found: 'luaotfload-main'
Error in luaotfload: reverting to OT1 L3 programming layer <2022-01-21>
(/usr/share/texlive/texmf-dist/tex/latex/base/article.cls
Document Class: article 2021/10/04 v1.4n Standard LaTeX document class
(/usr/share/texlive/texmf-dist/tex/latex/base/size10.clo
(/usr/share/texmf/tex/latex/lm/ot1lmr.fd))) (./lyluatex.sty

! LaTeX Error: File `luatexbase.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: 

! LaTeX Error: File `xkeyval.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: 
(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphicx.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics/keyval.sty)
(/usr/share/texlive/texmf-dist/tex/latex/graphics/graphics.sty
(/usr/share/texlive/texmf-dist/tex/latex/graphics/trig.sty)
(/usr/share/texlive/texmf-dist/tex/latex/graphics-cfg/graphics.cfg)
(/usr/share/texlive/texmf-dist/tex/latex/graphics-def/luatex.def)))

! LaTeX Error: File `minibox.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: 

! LaTeX Error: File `environ.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: 

! LaTeX Error: File `currfile.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: 

! LaTeX Error: File `pdfpages.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: 

! LaTeX Error: File `metalogo.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: 
[\directlua]:1: module 'lyluatex-options.lua' not found:
	no field package.preload['lyluatex-options.lua']
	[kpse lua searcher] file not found: 'lyluatex-options.lua'
	[kpse C searcher] file not found: 'lyluatex-options.lua'
	no file '/usr/local/lib/lua/5.3/lyluatex-options.so'
	no file '/usr/local/lib/lua/5.3/loadall.so'
	no file './lyluatex-options.so'
stack traceback:
	[C]: in function 'require'
	[\directlua]:1: in main chunk.
l.132 }
     
? 
./lyluatex.lua:12: module 'lyluatex-lib.lua' not found:
	no field package.preload['lyluatex-lib.lua']
	[kpse lua searcher] file not found: 'lyluatex-lib.lua'
	[kpse C searcher] file not found: 'lyluatex-lib.lua'
	no file '/usr/local/lib/lua/5.3/lyluatex-lib.so'
	no file '/usr/local/lib/lua/5.3/loadall.so'
	no file './lyluatex-lib.so'
stack traceback:
	[C]: in function 'require'
	./lyluatex.lua:12: in main chunk
	[C]: in function 'require'
	[\directlua]:1: in main chunk.
l.140 }
     
? 
! Undefined control sequence.
l.250 \NewEnviron
               {ly@compilely}{%
? 

! LaTeX Error: Missing \begin{document}.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.250 \NewEnviron{l
                 y@compilely}{%
? 
! Undefined control sequence.
\ly@compilescore ...'currfiledir', [[\currfiledir 
                                                  ]]) ly_opts:set_option('tw...

l.254   )}
        %
? 
! Undefined control sequence.
<argument> ly.fragment( '\luatexluaescapestring 
                                     {\unexpanded \expandafter {\BODY }}', [[\o
l.254   )}
        %
? 
! Undefined control sequence.
<argument> ...aescapestring {\unexpanded \expandafter {\BODY 
                                                  }}', [[\options ]] )
l.254   )}
        %
? 
! Undefined control sequence.
<argument> ...unexpanded \expandafter {\BODY }}', [[\options 
                                                  ]] )
l.254   )}
        %
? 
[\directlua]:1: attempt to index a nil value (global 'ly_opts')
stack traceback:
	[\directlua]:1: in main chunk.
\ly@compilescore ... #1 ly.newpage_if_fullpage() }
                                                  \ly@resetunits \ly@current...

l.254   )}
        %
? 
[\directlua]:1: attempt to index a nil value (global 'ly_opts')
stack traceback:
	[\directlua]:1: in main chunk.
\ly@currentfonts ..._font_family(font.current()))}
                                                  \rmfamily \edef \rmfamilyi...

l.254   )}
        %
? 
(/usr/share/texmf/tex/latex/lm/ot1lmss.fd)
(/usr/share/texmf/tex/latex/lm/ot1lmtt.fd)[\directlua]:1: attempt to index a ni
l value (global 'ly')
stack traceback:
	[\directlua]:1: in main chunk.
\ly@currentfonts ..., \sffamilyid , \ttfamilyid )}
                                                  \endgroup 
l.254   )}
        %
? 
[\directlua]:1: attempt to index a nil value (global 'ly')
stack traceback:
	[\directlua]:1: in main chunk.
\ly@compilescore ...directlua {ly.score:process()}
                                                  
l.254   )}
        %
? 
) (/usr/share/texlive/texmf-dist/tex/latex/l3backend/l3backend-luatex.def)
(./untitled-3.aux) (/usr/share/texlive/texmf-dist/tex/latex/base/ts1cmr.fd)
(/usr/share/texlive/texmf-dist/tex/context/base/mkii/supp-pdf.mkii
[Loading MPS to PDF converter (version 2006.09.02).]
) (/usr/share/texlive/texmf-dist/tex/latex/epstopdf-pkg/epstopdf-base.sty[\dire
ctlua]:1: module 'pdftexcmds' not found:
	no field package.preload['pdftexcmds']
	[kpse lua searcher] file not found: 'pdftexcmds'
	[kpse C searcher] file not found: 'pdftexcmds'
stack traceback:
	[C]: in function 'require'
	[\directlua]:1: in main chunk.
l.165       \directlua{require('pdftexcmds')}
                                           
? 
(/usr/share/texlive/texmf-dist/tex/latex/latexconfig/epstopdf-sys.cfg))
[1{/var/lib/texmf/fonts/map/pdftex/updmap/pdftex.map}] (./untitled-3.aux))
 368 words of node memory still in use:
   3 hlist, 1 vlist, 1 rule, 2 glue, 48 glue_spec, 1 write nodes
   avail lists: 2:13,3:4,4:1,5:22,6:2,7:58,9:24,11:2
{/usr/share/texmf/fonts/enc/dvips/lm/lm-rm.enc}</usr/share/texmf/fonts/type1/pu
blic/lm/lmr10.pfb>
Output written on untitled-3.pdf (1 page, 24173 bytes).
SyncTeX written on untitled-3.synctex.gz.
Transcript written on untitled-3.log.
```

And here is the [log file.](https://github.com/jperon/lyluatex/files/9742501/untitled-3.log)

**Comment by rpspringuel** [comment]
Now it looks like you've got some of the dependency sty files missing.  Which tex packages did you install on Debian?

**Comment by BetazoidMan** [comment]
Maybe I'm misunderstanding you, but I installed using using the 2022 version: `install-tl-20220815`. And I let it install everything; I didn't say to only install part. 

Thank you for your help.

**Comment by BetazoidMan** [comment]
At the advice of [this StackExchange article,](https://tex.stackexchange.com/questions/410949/latex-error-file-luaotfload-sty-not-found) I used the 'Search the contents of packages' search bar of [this ubuntu package database](https://packages.ubuntu.com/) to find the packages I was missing. Then I used `sudo apt-get install PACKAGENAME ` to install the packages I was missing. Now, it is compiling just fine.
